packages/react-devtools-shared/src/devtools/views/Components/KeyValue.js

```diff
@@ -264,7 +267,7 @@ export default function KeyValue({
         <div className={styles.ExpandCollapseToggleSpacer} />
         {renderedName}
         <div className={styles.AfterName}>:</div>
-        {canEditValues ? (
+        {!readOnly && canEditValues ? (
```

Comment:
Can we make this part of the `canEditValues` boolean assignment?

It doesn't make sense for `canEditValues` to be true if `readOnly` is also true, and that way we don't have to repeat the check everywhere below.

Reply:
Done. I also realized that I should apply `readOnly` to `canDeletePaths` and `canRenamePaths` as well.
Note that I introduced variables `canEditValuesProp` etc. (without `readOnly` applied) and pass those to the child `KeyValue` components because I wanted to be consistent about where `readOnly` is applied: if I apply it to the prop that I pass to child `KeyValue` components here, I should apply it to the prop of _all_ `KeyValue` components, i.e. I'd have to modify `InspectedElementContextTree`, `InspectedElementHooksTree` and `InspectedElementStateTree` as well...